src/lib/draw/events.ts

```diff
@@ -0,0 +1,42 @@
+import type { Map } from "maplibre-gl";
+
+// Use this to create and destroy event handlers on the map and document. When
```

Comment:
This is a bit fancier, for classes. We can't just say `map.on("click", this.onClickHandler)`, because we need to bind the current object as `this`. That binding step messes up calling `off` and having it match. There's maybe some simpler things we could do by defining all methods as arrow functions; see https://osmus.slack.com/archives/C01G3D28DAB/p1682086599507859 for discussion